packages/langium/test/parser/langium-parser-builder.test.ts

```diff
@@ -73,6 +73,37 @@ describe('Predicated grammar rules', () => {
 
 });
 
+describe('One name for terminal and non-terminal rules', () => {
+    let parseResult: ParseResult<Grammar>;
+    const content = `
+    grammar Test
+
+    entry Main: A | B | C;
+    
+    A: 'A' Bdata Cterm prop=B;
+    
+    B: Bdata Cterm 'A' prop=C;
+    Bdata returns string: 'B';
+    
+    C: Cterm 'A' Bdata prop=A;
+    terminal Cterm: /C/;    
+    `;
+
+    beforeAll(async () => {
+        parseResult = (await helper(content)).document.parseResult;
+        parserFromGrammar(parseResult.value);
+    });
+
+    test('Should have no lexer errors', () => {
+        expect(parseResult.lexerErrors).toEqual([]);
+    });
+
+    test('Should have no parser errors', () => {
+        expect(parseResult.parserErrors).toEqual([]);
+    });
```

Comment:
These tests don't actually do anything. Instead, the actual test should consist in the `parserFromGrammar` method not to throw:

```ts
test('Should not throw on parser validation', () => {
  expect(() => {
    parserFromGrammar(parseResult.value);
  }).not.toThrow();
});
```

Reply:
These tests fall, if I execute them on non-modified version and call `parserFromGrammar` [here](https://github.com/langium/langium/blob/0d5c796b2a9c4afa46121176b3344f1faf60a9a6/packages/langium/test/parser/langium-parser-builder.test.ts#L94), and I thought that they are working. Now I've got that any tests fall after `parserFromGrammar` executing, when it throws exeptions, so, my tests really do nothing. I rewrote them.